app/models/hostgroup.rb

```diff
@@ -16,6 +16,7 @@ class Hostgroup < ApplicationRecord
 
   validate :validate_subnet_types
   validates_with SubnetsConsistencyValidator
+  validate :validate_compute_profile
```

Comment:
You could also conditionally run this validation only when the attribute changed, e.g.:
```suggestion
  validate :validate_compute_profile, if: :compute_profile_id_changed?
```